internal/proxy/anthropic_hang_fixes_test.go

```
@@ -0,0 +1,247 @@
+
+	served := false
+	transport := replayablePostRetryTransport{
+		base: bedrockRoundTripFunc(func(*http.Request) (*http.Response, error) {
+			served = true
+			return &http.Response{StatusCode: http.StatusOK, Body: io.NopCloser(strings.NewReader(""))}, nil
+		}),
+		maxAttempts: 1,
+		limiter:     limiter,
+	}
+
+	small, err := http.NewRequest(http.MethodPost, "https://example.invalid/v1/messages", strings.NewReader(`{}`))
```

golangci-lint (2.12.2)

[error] net/http.NewRequest must not be called. use net/http.NewRequestWithContext

(noctx)